Modules/System/AL Rest Client/src/ALHttpRequestMessage.Codeunit.al

```diff
@@ -0,0 +1,86 @@
+/// <summary>Holder object for the HTTP request data.</summary>
+codeunit 50301 "AL Http Request Message"
+{
+    var
+        HttpRequestMessage: HttpRequestMessage;
+        Initialized: Boolean;
+        NotInitializedErr: Label 'The Http Request Message has been initialized';
+
+    /// <summary>Initializes the HttpRequestMessage object.</summary>
+    procedure Initialize()
+    begin
+        Clear(HttpRequestMessage);
+        Initialized := true;
+    end;
+
+    /// <summary>Sets the HTTP method or the HttpRequestMessage object.</summary>
+    /// <param name="Method">The HTTP method to use. Valid options are GET, POST, PATCH, PUT, DELETE, HEAD, OPTIONS</param>
+    /// <remarks>Default method is GET</remarks>
+    procedure SetHttpMethod(Method: Text)
+    begin
+        AssertInitialized();
+        HttpRequestMessage.Method := Method;
+    end;
+
+    /// <summary>Sets the HTTP method for the HttpRequestMessage object.</summary>
+    /// <param name="Method">The HTTP method to use.</param>
+    /// <remarks>Default method is GET</remarks>
+    procedure SetHttpMethod(Method: Enum "Http Request Type")
+    begin
+        SetHttpMethod(Method.Names.Get(Method.Ordinals.IndexOf(Method.AsInteger())));
+    end;
+
+    /// <summary>Sets the Uri used for the HttpRequestMessage object.</summary>
+    /// <param name="Uri">The Uri to use for the HTTP request.</param>
+    /// <remarks>The valued must not be a relative URI.</remarks>
+    procedure SetRequestUri(Uri: Text)
+    begin
+        AssertInitialized();
+        HttpRequestMessage.SetRequestUri(Uri);
+    end;
+
+    /// <summary>Adds a request header to the HttpRequestMessage object.</summary>
+    /// <param name="HeaderName">The name of the header to add.</param>
+    /// <param name="HeaderValue">The value of the header to add.</param>
+    /// <remarks>If the header already exists, it will be overwritten.</remarks>
+    procedure AddRequestHeader(HeaderName: Text; HeaderValue: Text)
```

Comment:
Since you're using this to add authorization headers in the AL Rest Client Impl. codeunit, should this procedure not be tagged as NonDebuggable? Alternatively, there can be a separate procedure, e.g. AddAuthorizationHeader(HeaderName: Text; HeaderValue: Text) that is tagged as NonDebuggable.

Reply:
Good point. I'll go with NonDebuggable on the AddRequestHeader because it can be used for other confidential information as well.